Assets/Scripts/DoorController.cs

```diff
@@ -47,4 +54,32 @@ public override uint GetUseTime(PlayerInteractionController Caller)
     {
         return interactionTime;
     }
+    private bool IsPlayerInBounds(GameObject Player)
+    {
+        BoxCollider2D playerCollider = Player.GetComponent<BoxCollider2D>();
```

Comment:
Potential null dereference, as object may potentially lack the collider.

Reply:
The only object that is going to use the door is the player, and player always has a collider, if it doesn't we made a mistake along the way. But I see your point, will fix